staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/conversion.go

```diff
@@ -0,0 +1,70 @@
+package v1alpha1
+
+import (
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+var _ runtime.NestedObjectDecoder = &AdmissionConfiguration{}
+
+// DecodeNestedObjects handles encoding RawExtensions on the AdmissionConfiguration, ensuring the
+// objects are decoded with the provided decoder.
+func (c *AdmissionConfiguration) DecodeNestedObjects(d runtime.Decoder) error {
+	// decoding failures result in a runtime.Unknown object being created in Object and passed
+	// to conversion
+	for k, v := range c.Plugins {
+		decodeNestedRawExtensionOrUnknown(d, &v.Configuration)
+		c.Plugins[k] = v
+	}
+	return nil
+}
+
+var _ runtime.NestedObjectEncoder = &AdmissionConfiguration{}
+
+// EncodeNestedObjects handles encoding RawExtensions on the AdmissionConfiguration, ensuring the
+// objects are encoded with the provided encoder.
+func (c *AdmissionConfiguration) EncodeNestedObjects(e runtime.Encoder) error {
+	for k, v := range c.Plugins {
+		if err := encodeNestedRawExtension(e, &v.Configuration); err != nil {
+			return err
+		}
+		c.Plugins[k] = v
+	}
+	return nil
+}
+
+func decodeNestedRawExtensionOrUnknown(d runtime.Decoder, ext *runtime.RawExtension) {
+	if ext.Raw == nil || ext.Object != nil {
```

Comment:
a little surprising that this func does nothing the second time it is called. This should at least be in the godoc.

Reply:
> a little surprising that this func does nothing the second time it is called. This should at least be in the godoc.

Ok.  The idea is that you've already decoded your object at the point and you update the object, not the raw

Reply:
> a little surprising that this func does nothing the second time it is called. This should at least be in the godoc.

@sttts added.